Fix DOM classList handling

loginx opened the pull request:
- When the DOMTokenList is empty on Chrome, it returns undefined
- This patch ensures the element classList can be resolved before using it

loginx committed:
Fix DOM classList handling

* When the DOMTokenList is empty on Chrome, it returns undefined
* This patch ensures the element classList can be resolved before using it

tiff commented:
Hey @loginx,

First of all thanks for your pull request!
Is there any documentation for this behavior in Chrome or can you elaborate under which circumstances this is happening? Sounds quite odd to me that the classList property becomes null.
Also your pull request misses a `var` statement. `classList` pollutes the global namespace.

loginx committed:
Add `var` statements to `api.*Class` handlers

loginx commented:
@tiff From what I can tell after looking into it some more, the `DOMTokenList` does _not_ become undefined on empty as I first thought.

The issue I'm running into is initializing a wysihtml5 editor instance that _is_ in the DOM, but not visible. As soon as that happens, I see a flurry of `api.removeClass` being called in an infinite loop against all the toolbar command buttons. It's trying to remove the `wysihtml5-command-disabled` class on the same elements continually, and eventually, even though the DOM elements exist, the element.classList reference is undefined.

Here's a screenshot of my dev tools console where I've added the following code to `api.removeClass()` right before the `element.classList.remove(className)` call:
`console.log('removeClass:', className, element.classList, element);`

![Screenshot](http://cl.ly/072P2j3v2Q073u1m0G3n/Screen%20Shot%202012-06-25%20at%2010.57.11%20AM.png)

The patch in this pull request resolves the errors by checking that classList exists first, but I think it only addresses the symptom and not the root cause. I've added the missing `var` statement to the methods as you suggested though.
